### PR TITLE
Updated output config file to be same as config file

### PR DESCRIPTION
**Pull request by theory-in-progress**
### Description
Updated the output `config-lock.yaml` to be the same as the user config file

---

### What is the purpose of this pull request?

- [ ] Bug fix
- [x] New feature
- [ ] Other

### CODE REVIEW

File: src/templates/template-common/utils.py [hunk 1]
```diff
@@ -144,6 +144,12 @@ def setup_output_dir(config: Any, rank: int) -> Path:
     return Path(idist.broadcast(config.output_dir, src=0))
 
 
+def write_config_to_output_dir(config: Any) -> None:
+    with open(config.config, "r") as f:
+        cfg = yaml.safe_load(f.read())
+        (config.output_dir / "config-lock.yaml").write_text(yaml.dump(cfg))
```
Comment (vfdev-5) [inline src/templates/template-common/utils.py:150]: Why can't we use shutil to copy the file ?

File: src/templates/template-vision-classification/main.py [hunk 2]
```diff
@@ -50,7 +49,7 @@ def run(local_rank: int, config: Any):
     # print training configurations
     logger = setup_logging(config)
     logger.info("Configuration: \n%s", pformat(vars(config)))
-    (config.output_dir / "config-lock.yaml").write_text(yaml.dump(config))
+    write_config_to_output_dir(config)
```
Comment (vfdev-5) [inline src/templates/template-vision-classification/main.py:52]: This comment is related to this PR and also to the previous code. How do we ensure that only one rank is writing to the disk avoiding data corruption if all ranks start to write into the same file together ?

Comment (vfdev-5) [reply src/templates/template-vision-classification/main.py:52]: Yes, but right now it is not yet the case, right ?

Comment (theory-in-progress) [reply src/templates/template-vision-classification/main.py]: Yes, made the changes!

**Comment by vfdev-5** [comment]
How about other templates ?

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**